python/social_bot/tasks.py

```diff
@@ -597,19 +648,41 @@ def task_specific_observation(self, agent):
             else:
                 pose = np.concatenate((pose, obj_pos), axis=0)
 
+        obj_array = []
         agent_pose = np.array(agent.get_pose()).flatten()
-        if self._use_egocentric_states:
+        if self._use_full_states or self._use_egocentric_states:
             yaw = agent_pose[5]
             # adds egocentric velocity input
             vx, vy, vz, a1, a2, a3 = np.array(agent.get_velocities()).flatten()
-            rvx, rvy = agent.get_egocentric_cord_2d(vx, vy, yaw)
-            obs = [rvx, rvy, vz, a1, a2, a3]
+            if self._use_egocentric_states:
+                rvx, rvy = agent.get_egocentric_cord_2d(vx, vy, yaw)
+            else:
+                rvx, rvy = vx, vy
+            obs = [rvx, rvy, vz, a1, a2, a3] + list(agent_pose)
+            if self._egocentric_perception_range > 0:
+                obj_array.extend(
+                    [[0, rvx, rvy, vz], [1, a1, a2, a3],
+                     [2, agent_pose[0], agent_pose[1], agent_pose[2]],
+                     [3, agent_pose[3], agent_pose[4], agent_pose[5]]])
+                i = 4
+            else:
+                obj_array.extend([[0, rvx, rvy], [1, vz, a1], [2, a2, a3],
+                                  [3, agent_pose[0], agent_pose[1]],
+                                  [4, agent_pose[2], agent_pose[3]],
+                                  [5, agent_pose[4], agent_pose[5]]])
+                i = 6
```

Comment:
This is very hacky.
A better way might be removing 662 - 673.  And in the loop starting from line 676, only fill obj_array, do not change obs. After that merge obs with obj_array.

Reply:
I've reverted this change.  The obj_array was added for adding an index to each object, so the objects can be selected using language instruction via e.g. GFT.  Removing as we are not pursuing that line of research at least for the moment.